### DISCUSSION TITLE
Stuck on MultiSaml Implementation

### DISCUSSION THREAD

**Discussion by Huzaifa-Asif** [bergie/passport-saml#877] [General]
I am implementing Multi Saml flow using this module but i am facing issues, i have worked with this library deprecated version and it worked fine but when i upgraded the code to this module i start facing errors. So if you have test implementation for multi saml kindly share the code it will be helpful. 

Plus i want to attach redis for cache management so if you have example of that how to integrate that in the flow it will be very helpful.

**Comment by srd90** [comment]
>...i have worked with this library deprecated version and it worked fine but when i upgraded the code to this module i start facing errors. ...

You might have better luck getting at least some help if you would describe what errors you are running into. Include also version numbers (of your old passport-saml version also).

>Plus i want to attach redis for cache management so if you have example of that how to integrate that in the flow it will be very helpful.

Do mean this well documented stuff for inresponseto validation (code snippet links point to version 4.0.4):
1. Implement this interface to wrap access to Redis
   https://github.com/node-saml/passport-saml/blob/0c2ccedf47076b0c8d68452eeb90af1465a20262/README.md?plain=1#L362-L385
2. And see this multisaml related note
    https://github.com/node-saml/passport-saml/blob/0c2ccedf47076b0c8d68452eeb90af1465a20262/README.md?plain=1#L107-L109

or do you mean something else?
